src/lm/searchTools.ts

```diff
@@ -0,0 +1,439 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+'use strict';
+
+import * as vscode from 'vscode';
+import Logger from '../common/logger';
+import { FolderRepositoryManager } from '../github/folderRepositoryManager';
+import { ILabel, Issue } from '../github/interface';
+import { RepositoriesManager } from '../github/repositoriesManager';
+import { ChatParticipantState } from './participants';
+import { concatAsyncIterable, ToolBase } from './tools/toolsUtils';
+
+interface ConvertToQuerySyntaxParameters {
+	naturalLanguageString: string;
+	repo?: {
+		owner?: string;
+		name?: string;
+	};
+}
+
+interface ConvertToQuerySyntaxResult {
+	query: string;
+	repo?: {
+		owner?: string;
+		name?: string;
+	};
```

Comment:
Is `repo: {}` valid?

Reply:
No, but it is valid for `SearchToolParameters` as the llm might not give us the proper arguments. Having `owner` and `name` optional here was the best way I could see to do this later:

https://github.com/microsoft/vscode-pull-request-github/blob/fe227313fea3d482a2cb42b15f6ce01cc1d36805/src/lm/searchTools.ts#L397-L398

which lets me describe that the output of one tool should be the input of the next.